### ISSUE TITLE
Provide End User estimated waiting for response

### ISSUE THREAD

**Issue by rasmusei** [buerokratt/Buerokratt-Chatbot#48]

**Comment by joonasroosalung** [comment]
@turnerrainer Is there a need for a separate waiting list table? It already exists in the sense that there is a chatlist, the queue is basically chats without assigned agents.

If there would a separate table, how would the cron update it and where would the cron run?

I'm halfway towards writing an sql query to calculate this. For example right now I can take timestamps from an active chat (that is a chat between customer and agent), between the last bot message and first agent message. This is the time the customer had to wait for the first reply from an agent.

So if you take those times, add them up for all customer-agent chats for example in the last 24h and divide them by the size of the queue, you get a waiting estimate. If you order the queue by created, the id is any chat's position in the queue.

What is easy: 
- getting the queue from chats without assigned agents, size and postion in queue
- finding waiting time (flow time)

Unknown/difficult:
- can it be done in one query, or could it be divided into multiple queries
- accuracy/usefulness of the estimate

Would this be worth pursuing, instead of the current dsl ac's?

**Comment by turnerrainer** [comment]
@joonasroosalung I agree, there's no need for a separate DB table for this as we are making these calculations only for live chats, not for a historical overview.

I'll update the AC.

**Comment by joonasroosalung** [comment]
@ceirowe Pushed mock yml to main as a placeholder, `/estimated-waiting-time` and `/estimated-waiting-time?chatId=any-value`. Using query params instead of path params as described in the original AC https://github.com/buerokratt/Ruuter/blob/main/samples/general/params.md.

**Comment by joonasroosalung** [comment]
The duration time will be in seconds from the last message of the bot to the final message in the chat. Will account chats that a) have a response from a human, b) have ended within the last month (hard-coded).

The average waiting time will be sum of the above duration times divided by the amount. Returned when no chatId parameter present in the request. Equivalent to being 1st in postion.

`{ "durationInSeconds": "100", positionInUnassignedChats: "" }`

When the request has the chatId parameter, will return the above average waiting time multiplied by the position, as well as the position itself.

`{ "durationInSeconds": "300", positionInUnassignedChats: "3" }`

**Comment by sergeirudz** [comment]
PR: https://github.com/buerokratt/Chat-Widget/pull/14

**Comment by ffrose** [comment]
BUG - 

1. The problem: Waiting for response(calculated time and queue number) does not exist.

2. How it should be: When atleast one CSA is available or idle and End User wants to talk with CSA, it displays calculated time and queue number so End User can decide if they want to wait or give their contact information and leave the chat.

3. When End User want to talk with CSA, it says "directing to CSA" without "Waiting for response" information.

How it is:
![image](https://github.com/buerokratt/Buerokratt-Chatbot/assets/119657383/31ccf8e0-b44f-4f74-8f00-8fac42192a98)


How it should be:
![image](https://github.com/buerokratt/Buerokratt-Chatbot/assets/119657383/cbcdb501-be21-4614-826b-6ffea99ce8b5)

**Comment by turnerrainer** [comment]
@rasmusei this one needs special attention. Help set the priorities.

**Comment by rasmusei** [comment]
Discussion with Rainer needed
